crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -135,11 +136,28 @@ impl Extension for QueryLimitsChecker {
         variables: &Variables,
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
+        let latency_timer = ctx
+            .data_opt::<Metrics>()
+            .map(|x| x.request_metrics.query_validation_latency.start_timer());
+        let timer_latency_by_path = ctx.data_opt::<Metrics>().map(|x| {
+            x.request_metrics
+                .query_validation_latency_by_path
+                // TODO make the path here
+                .with_label_values(&["path"])
```

Comment:
This path is easy to get for errors, but could not figure out yet how to do it for normal queries. Also, it's not clear to me how to make a path of a very complex query (multiple nested nodes, multiple sub-level queries).

Reply:
Errors in what sense?

We can extend the queue to also take in a string of its parents like "objects[n].owner.coins[n]"

Reply:
Leaving it out for now.